[dynamo] Store originating source in the Guard object

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107496
* #107645
* __->__ #107634
* #107622

Many times, I find myself wanting to know the source for the guard. This PR adds that as a field of guard itself.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov

anijain2305 committed:
[dynamo] Store originating source in the Guard object

[ghstack-poisoned]

voznesenskym commented on torch/_guards.py:
```diff
@@ -181,6 +184,14 @@ def inner_create_fn(self):
 
+    @property
+    def name(self):
+        return self.originating_source.name()
```
Cool :)

voznesenskym commented on torch/_guards.py:
```diff
@@ -181,6 +184,14 @@ def inner_create_fn(self):
+        return self.originating_source.name()
+
+    @property
+    def source(self):
```
this is a little confusing, and my fault, can you type annotate it `-> GuardSource:` ?

anijain2305 replied:
Yup, doing it.

anijain2305 committed:
Update on "[dynamo] Store originating source in the Guard object"


Many times, I find myself wanting to know the source for the guard. This PR adds that as a field of guard itself.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov

[ghstack-poisoned]
